### PR TITLE
fix publish_module_bundle crash

### PR DESCRIPTION
**Pull request by nkysg**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/move-language/move/blob/main/CONTRIBUTING.md#developer-workflow)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

File: language/move-core/types/src/vm_status.rs [hunk 1]
```diff
@@ -640,6 +640,8 @@ pub enum StatusCode {
     TOO_MANY_TYPE_NODES = 1116,
     VALUE_STACK_PUSH_OVERFLOW = 1117,
 
+    INVALID_EXTRACT_FOR_NATIVE_STRUCT = 1118,
```
Comment (jolestar) [inline language/move-core/types/src/vm_status.rs]: 加错误码容易冲突，这个是一个临时的解决方案，或者直接用一个已有的错误码，比如 `UNKNOWN_VERIFICATION_ERROR`？

File: language/tools/move-bytecode-utils/src/layout.rs [hunk 2]
```diff
@@ -175,7 +175,11 @@ impl<T: GetModule> SerdeLayoutBuilder<T> {
                         declaring_module.borrow().name()
                     )
                 });
-            let normalized_struct = Struct::new(declaring_module.borrow(), def).1;
+            let normalized_struct = Struct::new(declaring_module.borrow(), def);
```
Comment (jolestar) [inline language/tools/move-bytecode-utils/src/layout.rs]: 既然后面要 panic，不如这里直接 `expect("message")`

File: language/tools/move-cli/src/sandbox/utils/mod.rs [hunk 3]
```diff
@@ -343,7 +343,15 @@ pub(crate) fn explain_publish_error(
...
             let old_module = state.get_module_by_id(&module_id)?.unwrap();
             let old_api = normalized::Module::new(&old_module);
+            let old_api = match old_api {
+                Ok(old_api) => old_api,
+                e => return Err(format_err!("get module {:?} error {:?}", old_module, e))
+            };
             let new_api = normalized::Module::new(module);
+            let new_api = match new_api {
+                Ok(new_api) => new_api,
+                e => return Err(format_err!("get module {:?} error {:?}", module, e))
```
Comment (jolestar) [inline language/tools/move-cli/src/sandbox/utils/mod.rs]: 前面直接通过 `?` 返回？

**Review by jolestar — APPROVED** [review]

File: language/move-core/types/src/vm_status.rs (hunk 1 above)
Comment (nkysg) [reply language/move-core/types/src/vm_status.rs]: 我直接用UNKNOWN_VERIFICATION_ERROR

File: language/tools/move-bytecode-utils/src/layout.rs (hunk 2 above)
Comment (nkysg) [reply language/tools/move-bytecode-utils/src/layout.rs]: 我修改下

File: language/tools/move-cli/src/sandbox/utils/mod.rs (hunk 3 above)
Comment (nkysg) [reply language/tools/move-cli/src/sandbox/utils/mod.rs]: 这里是因为Module::new返回是Result<T, VMError>, 上面函数返回Result<()>, 直接？回报错

**Review by jolestar — APPROVED** [review]

**Pull request merged**